bad code

This code is buggy and does not compile. Errors in example 1, variable `hStdOut `is declared as `hStdout `..

The line: 
 _if (!WriteConsoleW(hStdOut, sequence, ARRAYSIZE(sequence), &written, NULL))_

does not compile (breaks on the macro ARRAYSIZE) .

error C2784: 'char (*RtlpNumberOf(__unaligned T (&)[N]))[N]': could not deduce template argument for '__unaligned T (&)[N]' from 'PCWSTR'

Did not test the other examples. Tested with Visual Studio 2019.

Edit: Here is my fixed version of example 1. 

```
#include <windows.h>
#include <string>

void CLS()
{
	HANDLE hStdOut;

	hStdOut = GetStdHandle(STD_OUTPUT_HANDLE);

	// Fetch existing console mode so we correctly add a flag and not turn off others
	DWORD mode = 0;
	if (!GetConsoleMode(hStdOut, &mode))
	{
		return;
	}

	// Hold original mode to restore on exit to be cooperative with other command-line apps.
	const DWORD originalMode = mode;
	mode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;

	// Try to set the mode.
	if (!SetConsoleMode(hStdOut, mode))
	{
		return;
	}

	// Write the sequence for clearing the display.
	DWORD written = 0;
	std::wstring sequence = L"\x1b[2J";
	if (!WriteConsoleW(hStdOut, sequence.c_str(), sequence.length(), &written, NULL))
	{
		// If we fail, try to restore the mode on the way out.
		SetConsoleMode(hStdOut, originalMode);
		return;
	}

	// To also clear the scroll back, emit L"\x1b[3J" as well.
	// 2J only clears the visible window and 3J only clears the scroll back.

	// Restore the mode on the way out to be nice to other command-line applications.
	SetConsoleMode(hStdOut, originalMode);

}
```
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 98b65d94-0ab7-d7ab-c309-cb883e93381d
* Version Independent ID: 09d1fad1-a5f6-22e2-2a78-9ffc970255ef
* Content: [Clearing the Screen - Windows Console](https://docs.microsoft.com/en-us/windows/console/clearing-the-screen)
* Content Source: [docs/clearing-the-screen.md](https://github.com/Microsoft/Console-Docs/blob/master/docs/clearing-the-screen.md)
* Product: **console**
* GitHub Login: @miniksa
* Microsoft Alias: **miniksa**

--- a/docs/clearing-the-screen.md
+++ b/docs/clearing-the-screen.md
@@ -29,11 +29,11 @@ The first method is to set your application up for virtual terminal output seque
 ```C
 #include <windows.h>
 
-int main( void )
+int main(void)
 {
-    HANDLE hStdout;
+    HANDLE hStdOut;
 
-    hStdout = GetStdHandle(STD_OUTPUT_HANDLE);
+    hStdOut = GetStdHandle(STD_OUTPUT_HANDLE);
 
     // Fetch existing console mode so we correctly add a flag and not turn off others
     DWORD mode = 0;
@@ -55,7 +55,7 @@ int main( void )
     // Write the sequence for clearing the display.
     DWORD written = 0;
     PCWSTR sequence = L"\x1b[2J";
-    if (!WriteConsoleW(hStdOut, sequence, ARRAYSIZE(sequence), &written, NULL))
+    if (!WriteConsoleW(hStdOut, sequence, (DWORD)wcslen(sequence), &written, NULL))
     {
         // If we fail, try to restore the mode on the way out.
         SetConsoleMode(hStdOut, originalMode);

